### DISCUSSION TITLE
I want to build 0.9.13 of carla, not 0.9.15

### DISCUSSION THREAD

**Discussion by goosebuums** [carla-simulator/carla#7319] [Q&A]
Hello, I have some trouble in building carla simulator in windows 11 environment. One thing really annoying is. I have to use Roadrunner of Mathworks, so I have to build carla 0.9.13. However, When I try to build in Unreal Engine, and check the plugin in menu, the carla version is always 0.9.15. I correctly extracted the asset file in Contentvesion.txt.
<img width="104" alt="image" src="https://github.com/carla-simulator/carla/assets/139331086/aad2375c-a80a-47dd-9392-967d74593dcd">
I've extracted the asset file in the correct location ,and when I make PythonAPI and make launch, 0.9.15 version of carla is installed. I have not entered the code, Update.bat, but the carla assets are updated automatically. I really want to solve this problem

**Comment by Jinviz** [comment]
Before installing an earlier version, you must install it as a clone from the repository, not from the plug-in installation in EpicGamesLauncher.

Enter 'git clone -b 0.9.13 https://github.com/carla-simulator/carla.git '  at the command prompt.

**Reply by goosebuums** [reply]
답변주셔서 감사합니다 
<img width="346" alt="image" src="https://github.com/carla-simulator/carla/assets/139331086/ce45a8e9-51ae-4b4e-9f57-49ef391c35e5">
이 코드 대신 말씀해주신 git clone -b 0.9.13 을 입력하면 된다는 말씀이신가요?

**Reply by Jinviz** [reply]
Yes, Include URL address to behind.

**Reply by Jinviz** [reply]
git clone -b 0.9.13 https://github.com/carla-simulator/carla.git

**Reply by Jinviz** [reply]
And you have to install a content file that fits version 0.9.13 instead of the latest content file

**Reply by Jinviz** [reply]
If it is successful, please let me know by adoption :)

**Reply by goosebuums** [reply]
Weird, it makes this error when I do "make PythonAPI"
C:\carla>make PythonAPI
-[Setup]: [Batch params]: --boost-toolset msvc-14.2 --all
-[Setup]: Asynchronous jobs:  32
-[Setup]: Boost toolset:      msvc-14.2
-[Setup]: Install directory:  "C:\carla\Build\"
-[Setup]: Installing zlib...
    -[install_zlib]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_zlib]: Not downloading zlib because already exists the folder "C:\carla\Build\zlib-source\".
CMake Error: The source directory "C:/carla/Build/zlib-source" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.

    -[install_zlib]: [CMAKE ERROR] An error ocurred while executing cmake command.
    -[install_zlib]: [CMAKE ERROR] Possible causes:
    -[install_zlib]:                - Make sure "CMake" is installed.
    -[install_zlib]:                - Make sure it is available on your Windows "path".
    -[install_zlib]:                - Make sure you have cmake 3.12.4 or higher installed.
    -[install_zlib]: Exiting with error...

-[Setup]:
   Ok, and error ocurred, don't panic!
   We have different platforms where you can find some help :)

   - Make sure you have read the documentation:
       http://carla.readthedocs.io/en/latest/how_to_build_on_windows/

   - If the problem persists, you can ask on our Github's "Building on Windows" issue:
       https://github.com/carla-simulator/carla/issues/21

   - Or just use our Discord channel!
       We'll be glad to help you there :)
       https://discord.gg/42KJdRj
make: *** [setup] Error 1

**Reply by goosebuums** [reply]
I've solved this problem. the problem was because zlib website is too old in this file.

**Reply by mohussein97** [reply]
hello which ubuntu and unreal versions are most suitable for .9.13 version and is the content file the same of assest file?

**Reply by sexiy** [reply]
> 我已经解决了这个问题。问题是因为 zlib 网站在此文件中太旧。

How was this problem solved? I also encountered this problem.

**Reply by Jinviz** [reply]
> 안녕하세요. .9.13 버전에 가장 적합한 Ubuntu 및 Unreal 버전은 무엇입니까? 그리고 콘텐츠 파일은 자산 파일과 동일합니까?

maybe UE4.26 and Contents file (2021ver)

**Reply by Samlitu** [reply]
> I've solved this problem. the problem was because zlib website is too old in this file.

Which "file" do you mean when you say "too old in this file" ?

Is it: .\carla\Util\InstallersWin\install_zlib.bat
Thanks in advance!

**Comment by mohussein97** [comment]
>And you have to install a content file that fits version 0.9.13 instead of the latest content file
